zerver/context_processors.py

```diff
@@ -236,3 +236,14 @@ def latest_info_context() -> Dict[str, str]:
         "latest_release_announcement": LATEST_RELEASE_ANNOUNCEMENT,
     }
     return context
+
+
+def find_accounts_email_context(user_profiles: UserProfile) -> Dict[str, Dict[str, Any]]:
+    context: Dict[str, Dict[str, Any]] = {}
+    for user in user_profiles:
+        context.setdefault(user.delivery_email, {})
+        context[user.delivery_email].setdefault("realms", [])
+        context[user.delivery_email]["realms"].append(user.realm)
+        context[user.delivery_email]["to_user_id"] = user.id
+
+    return context
```

Comment:
I think this function shouldn't exist and should just be a loop in the original function; `context_processors` is intended for common code used in multiple pages.